### PR TITLE
Fix (7531): Prioritise manual prices above oracles

### LINKED ISSUE #7531: Manual price is ignored if asset has special price calculation
## Problem Definition

If the asset has a special price calculation such as for curve pools, then we don't get to go to the manual price oracle, so any manually input price is ignored. So if our special price calculation method is broken, then there is no way for a user to fix it.

Seen in 1.32.1.

## Task

Fix this and make sure manual price oracle is preferred above all other methods.

### PR DESCRIPTION
**Pull request by OjusWiZard**
Closes #7531

## Checklist

- [x] Query manual price entries in DB before using the oracles

### CODE REVIEW

File: rotkehlchen/inquirer.py [hunk 1]
```diff
@@ -648,7 +668,24 @@ def _find_usd_price(
...
+                to_asset=A_USD,
+                coming_from_latest_price=coming_from_latest_price,
+                match_main_currency=match_main_currency,
+            )
+            Inquirer.set_cached_price(cache_key=cache_key, cached_price=CachedPriceEntry(
+                price=price,
+                time=ts_now(),
+                oracle=CurrentPriceOracle.MANUALCURRENT,
+                used_main_currency=used_main_currency,
+            ))
```
Comment (yabirgb) [inline rotkehlchen/inquirer.py:685]: can we add this to `_try_oracle_price_query`? If you check the other function we check if it is zero and if it is not then we log and add to the cache

File: rotkehlchen/inquirer.py [hunk 2]
```diff
@@ -648,7 +668,24 @@ def _find_usd_price(
...
+            return price, CurrentPriceOracle.MANUALCURRENT, used_main_currency
+
+        # continue, price can be found by one of the oracles (CC for example)
```
Comment (yabirgb) [inline rotkehlchen/inquirer.py:689]: No need for the extra space + this comment belongs right after querying the oracles, not here

File: rotkehlchen/inquirer.py [hunk 3]
```diff
@@ -648,7 +668,24 @@ def _find_usd_price(
                 price, oracle = Inquirer._query_fiat_pair(base=asset, quote=Inquirer.usd)
                 return price, oracle, False
 
-        # continue, asset isn't fiat or a price can be found by one of the oracles (CC for example)
+        # continue, asset isn't fiat, check manual prices
+        if GlobalDBHandler.get_manual_current_price(asset=asset) is not None:
+            price, used_main_currency, _ = Inquirer._try_oracle_price_query(
```
Comment (yabirgb) [inline rotkehlchen/inquirer.py:673]: Internally `Inquirer._manualcurrent` calls `GlobalDBHandler.get_manual_current_price(asset=asset)`. Should you just query `Inquirer._manualcurrent` and if the prize returned is not zero then return? Otherwise you are repeating a query

File: docs/changelog.rst [hunk 4]
```diff
@@ -2,6 +2,7 @@
 Changelog
 =========
 
+* :bug:`7531` rotki will now prioritize manual prices above other price oracles.
```
Comment (LefterisJP) [inline docs/changelog.rst:5]: Also mention other price calculations here.

File: rotkehlchen/tests/api/test_current_assets_price.py [hunk 5]
```diff
@@ -318,8 +318,8 @@ def test_manual_current_prices_loop(inquirer):
 
 @pytest.mark.vcr()
 @pytest.mark.parametrize('ignore_mocked_prices_for', ['ETH'])
-def test_inquirer_oracles_affect_manual_price(inquirer):
-    """Checks that change of oracles order affects manual current price usage.
+def test_inquirer_oracles_does_not_affect_manual_price(inquirer):
+    """Checks that change of oracles order does not affects manual current price usage.
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_current_assets_price.py:322]: Does not affect

File: rotkehlchen/inquirer.py [hunk 6]
```diff
@@ -411,7 +461,6 @@ def _query_oracle_instances(
         `coming_from_latest_price` is used by manual latest price oracle to handle price loops.
         """
         instance = Inquirer()
```
Comment (LefterisJP) [inline rotkehlchen/inquirer.py:463]: Why is that removed? Where did it go?

File: rotkehlchen/inquirer.py [hunk 7]
```diff
@@ -648,7 +672,17 @@ def _find_usd_price(
                 price, oracle = Inquirer._query_fiat_pair(base=asset, quote=Inquirer.usd)
                 return price, oracle, False
 
-        # continue, asset isn't fiat or a price can be found by one of the oracles (CC for example)
+        # continue, asset isn't fiat, check manual prices
+        if (price_result := Inquirer._try_oracle_price_query(
```
Comment (LefterisJP) [inline rotkehlchen/inquirer.py:676]: Hmm okay so you now run manual oracle before of o red it correctly ... But the manual oracle is reran in the for loop no? Didn't we say it should be removed during stand up?

File: rotkehlchen/inquirer.py (hunk 6 above)
Comment (OjusWiZard) [reply rotkehlchen/inquirer.py:463]: not sure, what you pointed but we moved some common code in `_try_oracle_price_query` function

Comment (LefterisJP) [reply rotkehlchen/inquirer.py:463]: I was pointing the line below. The cache key ... damn mobile app

File: rotkehlchen/tests/api/test_current_assets_price.py [hunk 8]
```diff
@@ -318,8 +318,8 @@ def test_manual_current_prices_loop(inquirer):
 
 @pytest.mark.vcr()
 @pytest.mark.parametrize('ignore_mocked_prices_for', ['ETH'])
-def test_inquirer_oracles_affect_manual_price(inquirer):
-    """Checks that change of oracles order affects manual current price usage.
+def test_inquirer_oracles_does_not_affect_manual_price(inquirer):
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_current_assets_price.py:321]: Avove this it is failing in `rotkehlchen/tests/api/test_current_assets_price.py::test_manual_current_prices_loop` with 

```
AssertionError: assert FVal(0) == FVal(1570.92) + where FVal(1570.92) = FVal('1570.92')
```

**Comment by LefterisJP** [comment]
Pushed a fix. 

```diff
modified   rotkehlchen/inquirer.py
@@ -435,6 +435,7 @@ class Inquirer:
                 f'Was not able to find price from {from_asset!s} to {to_asset!s} since your '
                 f'manual latest prices form a loop. For now, other oracles will be used.',
             )
+            is_error = False
 
         if (price, used_main_currency) != (ZERO_PRICE, False):
             Inquirer.set_cached_price(
```

I think that by moving the oracle price query out of the inquirer you ended up considering the manual price loop case as an error while it should not be and should be allowedto try other oracles.

Amended your commit with the diff above.

**Review by LefterisJP — APPROVED** [review]
LGTM. All seems green now
